### PR TITLE
fix: incorrect assertion func and json number parse rule

### PR DESCRIPTION
**Pull request by bbx-winner**
## Main Changes
1. fix incorrect assertion functions: greater_than, less_than, greater_or_equals, less_or_equals, adjust the sequence of parameters (actual and expected) of assertion function proto.
2. fix incorrect json number parse rule.
> Ref: [json.Unmarshal](https://pkg.go.dev/encoding/json#Unmarshal)
> 
> To unmarshal JSON into an interface value, Unmarshal stores one of these in the interface value:
> bool, for JSON booleans
> **float64, for JSON numbers**
> string, for JSON strings
> []interface{}, for JSON arrays
> map[string]interface{}, for JSON objects
> nil for JSON null
3. fix panic when Assertions map key not found.
4. fix unittest bad cases.

### CODE REVIEW

File: docs/BUILTIN.md [hunk 1]
```diff
@@ -31,16 +31,16 @@ Currently, HttpRunner+ has the following built-in assertion functions.
...
+| `len_lt`, `count_lt`, `length_less_than` | length less than | len(A) < B | 'abc' len_lt 4, [1,2,3] len_lt 4 |
+| `len_le`, `count_le`, `length_less_or_equals` | length less than or equals | len(A) <= B | 'abc' len_le 3, [1,2,3] len_le 3 |
 | `contains` | contains | [1, 2] contains 1 | 'abc' contains 'a', [1,2,3] len_lt 4 |
 | `contained_by` | contained by | A in B | 'a' contained_by 'abc', 1 contained_by [1,2] |
 | `type_match` | A and B are in the same type | type(A) == type(B) | 123 type_match 1 |
-| `regex_match` | regex matches | re.match(B, A) | 'abcdef' regex 'a\w+d' |
-| `startswith` | starts with | A.startswith(B) is True | 'abc' startswith 'ab' |
-| `endswith` | ends with | A.endswith(B) is True | 'abc' endswith 'bc' |
+| `regex_match` | regex matches | re.match(B, A) | 'abcdef' regex_match 'a\w+d' |
+| `starts_with` | starts with | A.starts_with(B) is True | 'abc' starts_with 'ab' |
```
Comment (debugtalk) [inline docs/BUILTIN.md:42]: 还是保留 startswith/endswith 吧；更自然些，而且跟 Python 版本保持一致

File: examples/httpbingo_test.json [hunk 2]
```diff
@@ -0,0 +1,81 @@
+{
+    "config": {
+        "name": "request methods testcase: validate with parameters",
+        "base_url": "https://httpbingo.bytedance.net"
```
Comment (debugtalk) [inline examples/httpbingo_test.json:4]: 代码中不要带字节的内网域名。示例代码还是使用 postman echo 的吧

Comment (bbx-winner) [reply examples/httpbingo_test.json]: fixed

File: docs/BUILTIN.md (hunk 1 above)
Comment (bbx-winner) [reply docs/BUILTIN.md]: fixed

**Review by debugtalk — APPROVED** [review]

**Pull request merged**